dit/dit.py

```diff
@@ -424,47 +439,52 @@ def _id_parse(self, argv):
 
     def _gname_parse(self, argv):
         names = argv.pop(0).split(self.separator)
+        if len(names) > 3:
+            raise Exception("Invalid gname format")
+        names = [name if name != self.root_name_cmd else self.root_name for name in names]
         names = names + [None] * (3 - len(names))
         group, subgroup, task = names
 
-        if group is None:
-            group = self.current_group
-        elif group == "":
-            group = None
-
-        if subgroup is None:
-            subgroup = self.current_subgroup
-        elif subgroup == "":
-            subgroup = None
-
         for name in [group, subgroup, task]:
-            if not self._is_valid_name(name):
-                raise Exception("Invalid name %s" % name)
+            if name and not self._is_valid_name(name):
+                raise Exception("Invalid name: %s" % name)
 
         return (group, subgroup, task)
 
     def _name_parse(self, argv):
         names = argv.pop(0).split(self.separator)
+        if len(names) > 3:
+            raise Exception("Invalid name format")
+        names = [name if name != self.root_name_cmd else self.root_name for name in names]
         names = [None] * (3 - len(names)) + names
+
         group, subgroup, task = names
 
         if group is None:
-            group = self.current_group
-        elif group == "":
-            group = None
+            if self.current_group is not None:
+                group = self.current_group
```

Comment:
`None` and `""` were already being treated differently in previous versions. That is not the point. The problem is that current_(sub)group is not consistent. What a current_(sub)group equals to None means? Currently, it means that you have never worked in any task, but makes no sense. I believe the initial value for current_(sub)group should be "" (the root group and subgroup). In that way, this method will always return some valid (sub)group and the checking is not necessary.

Why current_(sub)group is currently **inconsistent** and its initial state should not be `None`? In the current implementation, current_(sub)group is None if and only if you never worked in any task. That is, after your first `dit workon`, current_(sub)group will never be None again. Instead, it will be pointing to the last (sub)grouped you worked on. Now assume that you have created some tasks, worked and concluded all of them. There are now zero pending tasks, all of them are concluded. This state is similar to the initial one (not working in any task), but one is None and the other is not.